### PR TITLE
CIV-3107 free form docmosis

### PR DESCRIPTION
**Pull request by drummondjm**
https://tools.hmcts.net/jira/browse/CIV-3107
and 
https://tools.hmcts.net/jira/browse/CIV-6606

Docmosis changes for the free form final order journey

Does this PR introduce a breaking change? (check one with "x")

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/launchdarkly/FeatureToggleService.java
Comment (kurt-uk) [inline src/main/java/uk/gov/hmcts/reform/civil/launchdarkly/FeatureToggleService.java:1]: think youve added a file that was moved, youll need to remove this as its been moved to `src/main/java/uk/gov/hmcts/reform/civil/service/FeatureToggleService.java`

File: src/test/java/uk/gov/hmcts/reform/civil/service/docmosis/caseprogression/JudgeFinalOrderGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        when(documentManagementService
+                 .uploadDocument(BEARER_TOKEN, new PDF(fileFreeForm, bytes, JUDGE_FINAL_ORDER)))
+            .thenReturn(FREE_FROM_ORDER);
+
+        CaseData caseData = CaseDataBuilder.builder().atStateNotificationAcknowledged().build().toBuilder()
+            .finalOrderSelection(FinalOrderSelection.FREE_FORM_ORDER)
+            .build();
+        CaseDocument caseDocument = generator.generate(caseData, BEARER_TOKEN);
+
+        assertThat(caseDocument).isNotNull();
```
Comment (kurt-uk) [inline src/test/java/uk/gov/hmcts/reform/civil/service/docmosis/caseprogression/JudgeFinalOrderGeneratorTest.java:71]: could change these to the `assertNotNull()` method like lower down, keep with the junit assertions

**Review by kurt-uk — CHANGES_REQUESTED** [review]
can you also add on a screenshot of the docmosis template with some data in 😊

File: src/test/java/uk/gov/hmcts/reform/civil/service/docmosis/caseprogression/JudgeFinalOrderGeneratorTest.java (hunk 1 above)
Comment (drummondjm) [reply src/test/java/uk/gov/hmcts/reform/civil/service/docmosis/caseprogression/JudgeFinalOrderGeneratorTest.java]: And done

Also template is 
image

File: src/main/java/uk/gov/hmcts/reform/civil/launchdarkly/FeatureToggleService.java
Comment (drummondjm) [reply src/main/java/uk/gov/hmcts/reform/civil/launchdarkly/FeatureToggleService.java:1]: Done

File: src/test/java/uk/gov/hmcts/reform/civil/service/docmosis/caseprogression/JudgeFinalOrderGeneratorTest.java (hunk 1 above)
Comment (drummondjm) [reply src/test/java/uk/gov/hmcts/reform/civil/service/docmosis/caseprogression/JudgeFinalOrderGeneratorTest.java]: Note I raised some questions over <<claimantReference>> <<claimantReferenceTwo>> and the defendant equivalents, we're unclear on what it should be (believe we aim to clarify on PO review) and also that they are optional fields on create claim journeys, so potentially will never exist to be shown on template

File: src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/caseprogression/JudgeFinalOrderForm.java
Comment (bhagyashreesharma90) [inline src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/caseprogression/JudgeFinalOrderForm.java:1]: Can we move this class to model ?

Comment (drummondjm) [reply src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/caseprogression/JudgeFinalOrderForm.java:1]: Done

File: src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/caseprogression/JudgeFinalOrderGenerator.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                                            ? caseData.getOrderWithoutNotice().getWithoutNoticeSelectionTextArea() : null)
+            .withoutNoticeSelectionDate(nonNull(caseData.getOrderWithoutNotice())
+                                            ? caseData.getOrderWithoutNotice().getWithoutNoticeSelectionDate() : null);
+        return freeFormOrderBuilder.build();
+    }
+
+    // will be added part of civ-7569
+    private JudgeFinalOrderForm getAssistedOrder(CaseData caseData) {
+        return null;
+    }
```
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/caseprogression/JudgeFinalOrderGenerator.java:90]: I presume CIV-7569 will be worked on immediately afterwards?

Ideally I wouldn't want a placeholder implementation.

**Review by bhagyashreesharma90 — APPROVED** [review]

**Pull request merged**